### PR TITLE
Unnest ngrams

### PR DESCRIPTION
**Pull request by ColinFay**
This PR implement `unnest_ngrams()`, a wrapper around `unnest_tokens(token = "ngrams")`. 

Related to #137

### CODE REVIEW

**Comment by ColinFay** [comment]
Let me know if this PR sounds good, then the others will follow the same format :)

**Comment by EmilHvitfeldt** [comment]
I don't have any objections, I'm not a big fan of having `to_lower` so far back in the arguments, but the consistency with `unnest_tokens()` is better to preserve.

**Comment by juliasilge** [comment]
This is looking great! Thank you so much for your work on it. I'd like to see (or I can make) a couple of changes for these wrapper functions.

- Don't explicitly declare/pass any [tokenizers](https://github.com/ropensci/tokenizers) arguments that are not related to the specific tokenizer. In this case, that would be `simplify` and `stopwords`; instead pass the dots `...`, to leave flexibility for future changes, etc.
- Do explicitly document all `@param` items not inherited from `unnest_tokens()`. I'm looking particularly at, say, `n_min` which isn't clear enough as is. I guess an alternative is for one of us to submit a PR for the documentation to tokenizers so it says what it is more clearly.

File: R/unnest_ngrams.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  drop = TRUE,
+  collapse = NULL
+){
+  format <- match.arg(format)
+  unnest_tokens(tbl,
+                !! enquo(output),
+                !! enquo(input),
+                format = format,
+                to_lower = to_lower,
+                drop = to_lower,
```
Comment (juliasilge) [inline R/unnest_ngrams.R]: What's going on here with `drop = to_lower` and `collapse = to_lower`? Am I misunderstanding something, or is that just a typo?

Comment (ColinFay) [reply R/unnest_ngrams.R]: that's a typo, sorry. 

Corrected with the last commit

**Comment by ColinFay** [comment]
Hey @juliasilge, thanks for the feedbacks! 

Could you clarify what you'd like to change? 

> Don't explicitly declare/pass any tokenizers arguments that are not related to the specific tokenizer. In this case, that would be simplify and stopwords; instead pass the dots ..., to leave flexibility for future changes, etc.

Here, I've copied and pasted the args from [`tokenizers::tokenize_ngrams()`](https://github.com/ropensci/tokenizers/blob/master/R/ngram-tokenizers.R#L63) + `unnest_tokens` (so `simplify` and `stopwords` are there), how can I identify that an argument is not related to a tokenizer?

> Do explicitly document all @param items not inherited from unnest_tokens(). I'm looking particularly at, say, n_min which isn't clear enough as is. I guess an alternative is for one of us to submit a PR for the documentation to tokenizers so it says what it is more clearly.

Definitely agree that the documentation from `tokenizers` lacks some info there. It seems more logic to me to keep the two documentations synchronized, the `unnest_ngrams()` function passing arguments to `tokenize_ngrams()`, it seems legit to pass the documentation. I'm ok to change to an improved doc, though it seems that indeed it could benefit both `{tidytext}` and `{tokenizers}` to improve with a PR to `{tokenizers}`.

Colin

**Comment by juliasilge** [comment]
OK, for the issue with the `n_min` documentation, I will either try to submit a PR, or of course you @ColinFay are welcome to as well. Getting this PR here submitted and going does not depend on that. ✅ 

What I was getting at with the first point is that arguments like `simplify` and `stopwords` apply to many tokenizers and not specifically to the ngram tokenizer. It's not helpful specifically for n-gram tokenizing to know about the `simplify` argument. Adding in every possible argument explicitly makes these functions harder to parse, more overwhelming, and less aligned with the goal: wrapper functions that help people remember/use the arguments for specific kinds of tokenizing. 

Also, we do need to keep the `...` around, to keep flexibility for possible future changes in tokenizers. Since we're going to have `...` anyway, we should choose the arguments for `unnest_ngrams` that make the most sense for **that tokenizer**. For the n-gram tokenizer, that would be `n`, `n_min`, `ngram_delim`, and I'm pretty sure that's it?

Thank you so much for your patience, and for getting those typos fixed! 💪

**Comment by juliasilge** [comment]
Thank you so much @ColinFay for your work on this! 🎆 

I especially want to thank you for how patient you were, and I will work to be much more timely in the future on responding to these (if you want to submit any more) because I do think they are going to be very helpful for folks.

**Pull request merged**